packages/sdk/src/engine-commands/getConfig.ts

```diff
@@ -20,9 +20,9 @@ const unlink = promisify(fs.unlink)
 const MAX_BUFFER = 1_000_000_000
 
 export interface ConfigMetaFormat {
-  datasources: DataSource[] | []
-  generators: GeneratorConfig[] | []
-  warnings: string[] | []
+  datasources: DataSource[]
+  generators: GeneratorConfig[]
+  warnings: string[]
```

Comment:
Looks like a revert of a recent change (https://github.com/prisma/prisma/commit/52eed079145ee1ff8f04c8ed775a96da293a6477#diff-318f6444e319ef8973fda14740ee74061b8c384a0c7f06789a8105115aa62ecb)

Any reason to change this?

Reply:
It does not look like it's needed anymore? I see the inference issues are gone, but I might be wrong. Let me know :)

Reply:
About this I think it’s safer to revert and make it `DataSource[] | []`  since it was happens in reality
If you don’t revert this would be valid
`config.datasources[0]`
but could actually be undefined

Reply:
I discovered that in this commit https://github.com/prisma/prisma/commit/52eed079145ee1ff8f04c8ed775a96da293a6477
changed the type and then adapted to `config.datasources?.[0]`

Reply:
Thanks for the explanation! Fixed. We will try to add `noUncheckedIndexedAccess` globally via the tech debt initiative.